File: String/657.Robot_Return_to_Origin/C++/657.Robot_Return_to_Origin.cpp
class Solution {
public:
    bool judgeCircle(string moves) 
    {
        //===========================//
        // Input type:               //
        //   - string (moves string) //
        // Return type:              //
        //   - bool (result flag)    //
        //===========================//

        /*Initialize*/
        ///// Length of moves string /////
        int len_moves = moves.size();

        ///// Record hashmap, dictionary /////
        unordered_map<char, int> record_map;

        record_map['L'] = 0;
        record_map['R'] = 0;
        record_map['U'] = 0;
        record_map['D'] = 0;


        /*Simple differences & comparisons based loop traversal with recorded hashmap, dictionary*/
        ///// Step 1: Record indexed-char with hashmap, dictionary /////
        for (int moves_idx = 0; (moves_idx < len_moves); (moves_idx++)) //Whole
        {
            record_map[moves[moves_idx]] += 1; //Keep updating, recording

        } //Whole

        ///// Step 2: Check if the current hashmap matched conditions or not /////
        if ((record_map['L'] == record_map['R']) && (record_map['U'] == record_map['D'])) { return true; }

        return false;
    }
};

File: String/657.Robot_Return_to_Origin/Python/657.Robot_Return_to_Origin.py
class Solution(object):
    def judgeCircle(self, moves):
        """
        :type moves: str
        :rtype: bool
        """
        #===============================================#
        # Simple differences & comparisons based method #
        #===============================================#

        ############
        #Initialize
        ##### Length of moves string #####
        len_moves = len(moves)

        ##### Record dictionary/hashmap (i.e. left, right, up, down) #####
        record_dict = {'L': 0, 'R':0, 'U': 0, 'D': 0}


        ########################################################################################
        #Simple differences & comparisons based loop traversal with recorded dictionary/hashmap
        ##### Step 1: Record indexed-char with dictionary/hashmap #####
        for moves_idx in range(len_moves):
            record_dict[moves[moves_idx]] += 1 #Keep updating/recording

        ##### Step 2: Check if the current dictionary matched conditions or not #####
        if ((record_dict['L'] == record_dict['R']) and (record_dict['U'] == record_dict['D'])):
            return True

        return False
